### ISSUE TITLE
Neo Geo Battle Coliseum depth sort issue in per-triangle (Open GL/Vulkan)

### ISSUE THREAD

**Issue by RaduNastase** [libretro/flycast#741]
On Vulkan (android) the game has severe sprite prority issues. 
![Screenshot_20191215-123231](https://user-images.githubusercontent.com/18061144/70864609-5d3db980-1f5c-11ea-9b1a-6eeb52f5a05c.jpg)

Tested on OnePlus 7, default Core settings.

**Comment by barbudreadmon** [comment]
iirc the game is known for requiring per-pixel alpha sorting (OIT)

**Comment by ez2torta** [comment]
I had issues with this today on windows, can't remember my settings tho :(

**Comment by RaduNastase** [comment]
Yeah. Was hoping Vulkan would fix that on android.

**Comment by barbudreadmon** [comment]
> Was hoping Vulkan would fix that on android.

As explained in the vulkan flycast news, OIT is something natively happening on dreamcast/naomi. The only difference with vulkan is that OIT is now available on android (while it was totally impossible with gles), but you still need to enable it, and it will still require more resources than non-OIT.

**Comment by RaduNastase** [comment]
So...can we enable it atm?

**Comment by barbudreadmon** [comment]
Yes, set alpha sorting in core option to per-pixel, you'll need some beefy android device though, which is why it's not enabled by default.

**Comment by flyinghead** [comment]
Per-pixel should run fine on a OnePlus 7. Just make sure to not upscale too much. Start at 640x480.

**Comment by RaduNastase** [comment]
Per Pixel indeed fixes Battle Coliseum. The setting is more taxing indeed (Capcom vs Snk 2 with it enabled, sees a split second fps drop when using a lot of effects on screen) but games still run comfortably full speed.

Per pixel alpha introduces however a new issue. In Capcom vs Snk 1, the character order select is all messed up like seen in the two pics.
![Screenshot_20191216-144850](https://user-images.githubusercontent.com/18061144/70910546-9a678180-2018-11ea-90bd-5715898a0dfc.jpg)
![Screenshot_20191216-144446](https://user-images.githubusercontent.com/18061144/70910547-9b001800-2018-11ea-8016-e1db0e2962a9.jpg)

On the bright side, this same setting fixes the old "KO" bug. Now it's as it should be:
![Screenshot_20191216-144728](https://user-images.githubusercontent.com/18061144/70910644-cc78e380-2018-11ea-917e-e9f7ae9a99d1.jpg)

**Comment by flyinghead** [comment]
This is a known issue on Adreno with per-pixel. Probably a driver bug or edge case. I'm going to push a workaround for now.

**Comment by RaduNastase** [comment]
Great. Let me know when it is integrated in the core. Can't wait to test it

**Comment by flyinghead** [comment]
Here https://github.com/libretro/flycast/commit/399d7305c50330e992e2510905bce1b67139cc59

**Comment by RaduNastase** [comment]
Thanks, but what do i do with this, man?

**Comment by flyinghead** [comment]
That's the version of the core you want

**Comment by RaduNastase** [comment]
Apologies for not being more into the technical part of things. I have no idea what to do with that version of the core. I always update cores in my phone, via Retroarch GUI.

**Comment by flyinghead** [comment]
No problem. Just update the core as you normally do. It should be available now, that or a later version.

**Comment by RaduNastase** [comment]
Ok, core updated!

CVS1 character order select bug is fixed. I'd say, atm CVS1 (with Alpha Sorting per-pixel) and CVS2 (it works just as wel with per-triangle), are 99% close to perfect in Flycast. I see no bugs whatsoever, the only issue remaining the slight choppiness in between menus.

Neo Geo Battle Coliseum is very good as well graphically (perfect maybe), though there is a split second drop in performance when certain effects/special moves are performed, but only the first time it's performed, then it's all fairly smooth, even if i do the same moves again, even in different fights.
